### PR TITLE
Fix temperature scaling of vLLM logprobs

### PR DESCRIPTION
**Pull request by mikasenghaas**
> This PR is based on #696 and should not be merged before

This PR fixes a bug in our temperature-scaling of raw vLLM logprobs. Previously we did `logprobs /= temperature` for scaling, which is just plain wrong. Temperature scaling is applied before the softmax, but because we only have access to chosen token logprobs, we cannot reverse this process. Luckily, v0.10.0 allows us to specify processed logprobs via `--logprob-mode processed_logprobs` which we now set as a default, and raise if any other value is set . Funnily, vLLM's definition of "processed" does not include temperature scaling so we also have to monkeypatch the `Sampler` class to apply temperature scaling before (not after) exporting the `processed_logprobs` . With this we get the expected behavior of the importance ratio at lower temperature levels for `hendrycks-math`

Screenshot 2025-08-08 at 5 39 06 PM

All runs are 10 steps on `hendrycks-math` with 1B model with no recompute, ie. using vLLM logprobs:
- Green: How the importance ratio should look (around 1) at temperature 1.0 (baseline)
- Pink: What we have on current main, completely wrong temperature scaling (we scaled logprobs, not logits oops)
- Blue: After bump to v0.10.0 exproting `processed_logprobs` (new vllm feature), still off because "processed" does not include temperature scaling in vllm's mind (wtf?)
- Orange: After a one-line patch to the sampler we are finally good

### CODE REVIEW

File: src/prime_rl/trainer/train.py [hunk 1]
```diff
@@ -213,7 +213,6 @@ def train(config: TrainerConfig):
                     shifted_logits = shift_logits(logits)
                     shifted_logits = shifted_logits / temperature
                     recomputed_logprobs = selective_log_softmax(shifted_logits, input_ids)
-                    del logits, shifted_logits
```
Comment (Jackmin801) [inline src/prime_rl/trainer/train.py:216]: why dont we need this anymore? would this not save memory?

Comment (mikasenghaas) [reply src/prime_rl/trainer/train.py:216]: yes but i figured we don't really care in the logprob recomputation loop because our mem spike is during .backward() in the main grad acc loop, and we use the same variables `logits` and `shifted_keys` which we del before backward

**Comment by mikasenghaas** [comment]
> lgtm! would be nice to PR moving the apply_temperature up to vllm. guessing this is useful for many

You mean PRing this into vLLM?
